plugins/destination/gremlin/client/client.go

```diff
@@ -0,0 +1,122 @@
+package client
+
+import (
+	"context"
+	"crypto/tls"
+	"fmt"
+	"net/http"
+	"strings"
+	"sync"
+	"time"
+
+	gremlingo "github.com/apache/tinkerpop/gremlin-go/v3/driver"
+	v4 "github.com/aws/aws-sdk-go-v2/aws/signer/v4"
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/cloudquery/plugin-sdk/plugins/destination"
+	"github.com/cloudquery/plugin-sdk/specs"
+	"github.com/rs/zerolog"
+)
+
+type Client struct {
+	destination.UnimplementedUnmanagedWriter
+	destination.DefaultReverseTransformer
+	logger     zerolog.Logger
+	spec       specs.Destination
+	pluginSpec Spec
+
+	mu     sync.Mutex // protects client during session creation
+	client *gremlingo.DriverRemoteConnection
+}
+
+var AnonT = gremlingo.T__
+
+func New(ctx context.Context, logger zerolog.Logger, destSpec specs.Destination) (destination.Client, error) {
+	var err error
+	c := &Client{
+		logger: logger.With().Str("module", "gremlin").Logger(),
+		spec:   destSpec,
+	}
+	var spec Spec
+	if err := destSpec.UnmarshalSpec(&spec); err != nil {
+		return nil, fmt.Errorf("failed to unmarshal gremlin spec: %w", err)
+	}
+	spec.SetDefaults()
+	if err := spec.Validate(); err != nil {
+		return nil, err
+	}
+
+	c.pluginSpec = spec
+	u := spec.Endpoint + "/gremlin"
+
+	au, err := c.getAuthInfo(ctx, u)
+	if err != nil {
+		return nil, err
+	}
+
+	c.client, err = gremlingo.NewDriverRemoteConnection(u,
+		func(settings *gremlingo.DriverRemoteConnectionSettings) {
+			settings.TraversalSource = "g"
+			settings.AuthInfo = au
+
+			if logger.GetLevel() <= zerolog.DebugLevel {
+				settings.LogVerbosity = gremlingo.Debug
+			}
+
+			if spec.Insecure {
+				settings.TlsConfig = &tls.Config{InsecureSkipVerify: true}
+			}
+			settings.Logger = &Logger{Base: logger.With().Str("from", "gremlingo").Logger()}
+		})
+	if err != nil {
+		return nil, err
+	}
+
+	return c, nil
+}
+
+func (c *Client) Close(_ context.Context) error {
+	c.client.Close()
+	return nil
+}
+
+func (c *Client) getAuthInfo(ctx context.Context, baseURL string) (*gremlingo.AuthInfo, error) {
+	switch c.pluginSpec.AuthMode {
+	case authModeNone:
+		return nil, nil
+	case authModeBasic:
+		return gremlingo.BasicAuthInfo(c.pluginSpec.Username, c.pluginSpec.Password), nil
+
+	case authModeAWS:
+		// emptyStringSHA256 is a SHA256 of an empty string
+		const emptyStringSHA256 = `e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855`
+
+		req, err := http.NewRequest(http.MethodGet, baseURL, strings.NewReader(""))
+		if err != nil {
+			return nil, err
+		}
+
+		cfg, err := config.LoadDefaultConfig(ctx, config.WithDefaultRegion("us-east-1"))
+		if err != nil {
+			return nil, fmt.Errorf("unable to load AWS SDK config: %w", err)
+		}
```

Comment:
I don't think this should be `us-east-1`... I think it should be `cfg.Region` otherwise the SDK will use the us-east-1 sts endpoint rather than the local endpoint that is in the same region as the neptune db

Reply:
"Default Region" is just that, if there's a `region` specified in the config (or an `AWS_REGION` env var) it would be used instead? That's my thinking but could be mistaken

Reply:
I agree with your thinking... sorry, I misunderstood/misread `cfg` ... I was thinking it came from the `spec` rather than the `cfg` for the aws sdk...

Reply:
I made it simpler (or easier/clearer) and put in an `aws_region` field in the spec.